reader/views.py

```diff
@@ -2199,13 +2200,23 @@ def calendars_api(request):
         import datetime
         diaspora = request.GET.get("diaspora", "1")
         custom = request.GET.get("custom", None)
+
+        zone = request.GET.get("timezone", None)
+        if zone:
+            try:
+                zone = pytz.timezone(zone)
+            except pytz.exceptions.UnknownTimeZoneError as e:
+                return jsonResponse({"error": "Unknown 'timezone' value: '%s'." % zone})\
+        else:
+            zone = timezone.now()
+
         try:
             year = int(request.GET.get("year", None))
             month = int(request.GET.get("month", None))
             day = int(request.GET.get("day", None))
-            datetimeobj = datetime.datetime(year, month, day)
+            datetimeobj = datetime.datetime(year, month, day, tzinfo=zone)
         except Exception as e:
-            datetimeobj = timezone.localtime(timezone.now())
+            datetimeobj = timezone.localtime(zone)
```

Comment:
Looks like the `timezone` param needs to be specified since it's the 2nd parameter
```suggestion
            datetimeobj = timezone.localtime(timezone=zone)
```